src/cdk/datetime/datepicker.spec.ts

```diff
@@ -0,0 +1,646 @@
+import {FormControl, FormsModule, ReactiveFormsModule} from '@angular/forms';
+import {By} from '@angular/platform-browser';
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+import {CdkDatepicker} from './datepicker';
+import {CdkDatepickerInput} from './datepicker-input';
+import {CdkDatepickerModule} from './index';
+import {BrowserDynamicTestingModule} from '@angular/platform-browser-dynamic/testing';
+
+describe('CdkDatepicker', () => {
+  // Creates a test component fixture.
+  function createComponent(
+      component: Type<any>,
+      imports: Type<any>[] = [],
+      providers: (FactoryProvider | ValueProvider)[] = [],
+      entryComponents: Type<any>[] = []): ComponentFixture<any> {
+
+    TestBed.configureTestingModule({
+      imports: [
+        FormsModule,
+        CdkDatepickerModule,
+        NoopAnimationsModule,
+        ReactiveFormsModule,
+        ...imports
+      ],
+      providers,
+      declarations: [component, ...entryComponents],
+    });
+
+    TestBed.overrideModule(BrowserDynamicTestingModule, {
+      set: {
+        entryComponents: [entryComponents]
+      }
+    }).compileComponents();
+
+    return TestBed.createComponent(component);
+  }
+
+  describe('with NativeDateModule', () => {
+    describe('standard datepicker', () => {
+      let fixture: ComponentFixture<StandardDatepicker>;
+      let testComponent: StandardDatepicker;
+
+      beforeEach(fakeAsync(() => {
+        fixture = createComponent(StandardDatepicker, [NativeDateModule]);
+        fixture.detectChanges();
+
+        testComponent = fixture.componentInstance;
+      }));
+
+      afterEach(fakeAsync(() => {
+        fixture.detectChanges();
+        flush();
+      }));
+
+      it('should initialize with correct value shown in input', () => {
+        expect(fixture.nativeElement.querySelector('input').value).toBe('1/1/2020');
+      });
+
+      it('startAt should fallback to input value', () => {
+        expect(testComponent.datepicker.startAt).toEqual(new Date(2020, JAN, 1));
+      });
+
+      it('should not throw when given wrong data type', () => {
+        testComponent.date = '1/1/2017' as any;
+
+        expect(() => fixture.detectChanges()).not.toThrow();
+      });
+    });
+
+    describe('datepicker with too many inputs', () => {
+      it('should throw when multiple inputs registered', fakeAsync(() => {
+        let fixture = createComponent(MultiInputDatepicker, [NativeDateModule]);
+        expect(() => fixture.detectChanges()).toThrow();
+      }));
+    });
+
+    describe('datepicker with no inputs', () => {
+      let fixture: ComponentFixture<NoInputDatepicker>;
+      let testComponent: NoInputDatepicker;
+
+      beforeEach(fakeAsync(() => {
+        fixture = createComponent(NoInputDatepicker, [NativeDateModule]);
+        fixture.detectChanges();
+
+        testComponent = fixture.componentInstance;
+      }));
+
+      afterEach(fakeAsync(() => {
+        fixture.detectChanges();
```

Comment:
This seems a little weird. Why would we need to trigger change detection after the test is done? Same goes for the other similar `afterEach` blocks.